GKE-runner/variables.tf

```diff
@@ -0,0 +1,376 @@
+}
+
+variable "runner_node_pool_zones" {
+  type        = list(string)
+  description = "The zones to host the cluster in (optional if regional cluster / required if zonal)"
+  default     = null
+}
+
+variable "runner_node_pool_image_type" {
+  type        = string
+  description = "(optional) The type of image to be used"
+  default     = "COS"
+}
+
+variable "runner_node_pool_disk_size_gb" {
+  default     = 30
+  description = "(Optional) Size of the disk attached to each node, specified in GB. The smallest allowed disk size is 10GB"
+}
+
+variable "runner_node_pool_disk_type" {
+  default     = "pd-standard"
+  description = "(Optional) Type of the disk attached to each node (e.g. 'pd-standard', 'pd-balanced' or 'pd-ssd')."
+}
+
+variable "prefix" {
+  description = "string to be prepended to the nodes service account id and the service account for the cache"
+  type        = string
+  default     = "gitlab-runner"
+}
+
+variable "additional_node_service_account_roles" {
+  description = "additional roles to grant the service account"
+  type        = list(any)
+  default     = []
+}
+
+variable "runner_node_pool_name" {
+  description = "name of the runner node pool"
+  type        = string
+  default     = null
+}
+
+variable "initial_node_count" {
+  description = "initial number of nodes that the node pool creates"
+  type        = number
+  default     = 0
+}
+
+variable "runner_node_pool_min_node_count" {
+  description = "the minimum number of nodes that can be present in the node pool (autoscaling controls)"
+  type        = number
+  default     = 0
+}
+
+variable "runner_node_pool_max_node_count" {
+  description = "the maximum number of nodes that can be present in the node pool (autoscaling controls)"
+  type        = number
+  default     = 5
+}
+
+variable "runner_node_pool_machine_type" {
+  description = "type of compute machine used for the nodes in the runner node pool"
+  type        = string
+  default     = "n1-standard-2"
+}
+
+variable "runner_node_pool_node_labels" {
+  description = "labels for nodes in the runner node pool"
+  type        = map(any)
+  default = {
+    "role" = "gitlab-runner"
+  }
+}
+
+variable "runner_node_pool_node_taints" {
+  description = "taints to be applied to the nodes in the runner node pool"
+  type        = list(map(string))
+  default = [{
+    effect = "NO_SCHEDULE"
+    key    = "role"
+    value  = "gitlab-ci"
+  }]
+}
+
+# variable "node_is_preemptible" {
+#   default     = false
+#   type        = bool
+#   description = "A boolean that represents whether or not the underlying node VMs are preemptible"
+# }
+
+variable "runner_node_pool_oauth_scopes" {
+  description = "(Optional) Scopes that are used by NAP when creating node pools."
+  default     = ["https://www.googleapis.com/auth/cloud-platform"]
+  type        = list(string)
+}
+
+variable "cache_location" {
+  description = "location of the cache bucket"
+  type        = string
+  # default     = null
```

Comment:
is this a required variable? if region is specified, do we need to force them to pass `cache_location`?